user_modules/shared_testcases/commonFunctions.lua

```diff
@@ -951,11 +961,190 @@ end
+local function fill_request_parameters(vehicle_data)
+  local request_parameters={}
+  for _, v in pairs(vehicle_data) do
+    request_parameters[v] = true
+  end
+  return request_parameters
+end
+
+--------------------------------------------------------------------------------------------
+--Function fills response parameters for subscribe unsubscrive vehicle data
+---------------------------------------------------------------------------------------------
+--! @brief Fills table with parameter for subscribe unsubscribe vehicle data response 
+--! @param vehicle_data contains name of parameters
+--! @param result_codes contains table with result code for vehicle data params
+--! @return Returns table with response parameters
+local function fill_response_parameters(vehicle_data, result_codes)
+  local SVDValues = {gps           = "VEHICLEDATA_GPS",
+    speed         = "VEHICLEDATA_SPEED",
+    rpm           = "VEHICLEDATA_RPM",
+    fuelLevel       = "VEHICLEDATA_FUELLEVEL",
+    fuelLevel_State     = "VEHICLEDATA_FUELLEVEL_STATE",
+    instantFuelConsumption  = "VEHICLEDATA_FUELCONSUMPTION",
+    externalTemperature   = "VEHICLEDATA_EXTERNTEMP",
+    prndl         = "VEHICLEDATA_PRNDL",
+    tirePressure      = "VEHICLEDATA_TIREPRESSURE",
+    odometer        = "VEHICLEDATA_ODOMETER",
+    beltStatus        = "VEHICLEDATA_BELTSTATUS",
+    bodyInformation     = "VEHICLEDATA_BODYINFO",
+    deviceStatus      = "VEHICLEDATA_DEVICESTATUS",
+    driverBraking     = "VEHICLEDATA_BRAKING",
+    wiperStatus       = "VEHICLEDATA_WIPERSTATUS",
+    headLampStatus      = "VEHICLEDATA_HEADLAMPSTATUS",
+    engineTorque      = "VEHICLEDATA_ENGINETORQUE",
+    accPedalPosition    = "VEHICLEDATA_ACCPEDAL",
+    steeringWheelAngle    = "VEHICLEDATA_STEERINGWHEEL",
+    vin           = "VEHICLEDATA_VIN"
+  }
+  local response_parameters={}
+  for _, v in pairs(vehicle_data) do
+    if  v == "clusterModeStatus" then
+      response_parameters["clusterModes"] = {
+          resultCode = result_codes[v],
+          dataType = SVDValues[v]
+      }
+    else
+      response_parameters[v] = {
+        resultCode = result_codes[v],
+        dataType = SVDValues[v]
+    }
+    end
+  end
+  return response_parameters
+end
+
+-- ---------------------------------------------------------------------------------------------
+-- Function Subscribe/Unsubscribe sdl to vehicle data
+-- ---------------------------------------------------------------------------------------------
+--! @brief  Subscribe/Unsubscribe sdl to vehicle data
+--! @param self contains Test
+--! @param vehicle_data contains name of parameters
+--! @param result_codes contains table with result code for vehicle data params
+--! @param hmi_function_id contains function id for request to HMI
+--! @param mob_function_id contains function id for request from mobile
+local function sub_unsub_vehicle_data(self, vehicle_data, result_codes, hmi_function_id, mob_function_id)
```

Comment:
@LuxoftAKutsan I use common function for internal domestic needs. This method is used from "subscribe_to_vehicle_data", "unsubscribe_to_vehicle_data". These method are visible outside.